public/scripts/extensions/tts/kokoro.js

```diff
@@ -0,0 +1,203 @@
+            }
+            this.ready = true;
+            return true;
+        } catch (error) {
+            console.error("Kokoro TTS initialization failed:", error);
+            this.ready = false;
+            return false;
+        }
+    }
+
+    async onRefreshClick() {
+        return await this.checkReady();
+    }
+
+    get settingsHtml() {
+        return `
+            <div class="kokoro_tts_settings">
+                <label for="kokoro_model_id">Model ID:</label>
+                <input id="kokoro_model_id" type="text" class="text_pole" value="${this.settings.modelId}" />
+
+                <label for="kokoro_dtype">Data Type:</label>
+                <select id="kokoro_dtype" class="text_pole">
+                    <option value="q8" ${this.settings.dtype === 'q8' ? 'selected' : ''}>q8 (Recommended)</option>
+                    <option value="fp32" ${this.settings.dtype === 'fp32' ? 'selected' : ''}>fp32 (High Precision)</option>
+                    <option value="fp16" ${this.settings.dtype === 'fp16' ? 'selected' : ''}>fp16</option>
+                    <option value="q4" ${this.settings.dtype === 'q4' ? 'selected' : ''}>q4 (Low Memory)</option>
+                    <option value="q4f16" ${this.settings.dtype === 'q4f16' ? 'selected' : ''}>q4f16</option>
+                </select>
+
+                <label for="kokoro_device">Device:</label>
+                <select id="kokoro_device" class="text_pole">
+                    <option value="wasm" ${this.settings.device === 'wasm' ? 'selected' : ''}>WebAssembly (CPU)</option>
+                    <option value="webgpu" ${this.settings.device === 'webgpu' ? 'selected' : ''}>WebGPU (GPU Acceleration)</option>
+                </select>
+
+                <label for="kokoro_default_voice">Default Voice:</label>
+                <select id="kokoro_default_voice" class="text_pole">
+                    ${this.voices.map(voice =>
+                        `<option value="${voice}" ${this.settings.defaultVoice === voice ? 'selected' : ''}>${this.getVoiceDisplayName(voice)}</option>`
+                    ).join('')}
+                </select>
+
+                <label for="kokoro_speaking_rate">Speaking Rate: <span id="kokoro_speaking_rate_output">${this.settings.speakingRate}x</span></label>
+                <input id="kokoro_speaking_rate" type="range" value="${this.settings.speakingRate}" min="0.5" max="2.0" step="0.1" />
+
+                <label for="kokoro_volume_gain">Volume Gain: <span id="kokoro_volume_gain_output">${this.settings.volumeGainDb}dB</span></label>
+                <input id="kokoro_volume_gain" type="range" value="${this.settings.volumeGainDb}" min="-10" max="10" step="0.5" />
+
+                <hr>
+                <div class="kokoro_status" style="text-align: center;">
+                    <span id="kokoro_status_text">Not Initialized</span>
+                </div>
+            </div>
```

Comment:
Bigger inline HTML can much better be its own template file, and then loaded via `renderExtensionTemplateAsync`.